RUFAS/biophysical/animal/ration/user_defined_ration_manager.py

```diff
@@ -0,0 +1,104 @@
+from typing import Any
+
+from RUFAS.biophysical.animal.data_types.nutrition_data_structures import NutritionRequirements
+from RUFAS.data_structures.feed_storage_to_animal_connection import RUFAS_ID
+from RUFAS.enums import AnimalCombination
+from RUFAS.general_constants import GeneralConstants
+from RUFAS.output_manager import OutputManager
+from RUFAS.units import MeasurementUnits
+
+
+class UserDefinedRationManager:
+    """
+    Handles the initialization and management of user-defined animal rations.
+
+    Attributes
+    ----------
+    user_defined_rations : dict[AnimalCombination, dict[RUFAS_ID, float]]
+        Map of animal groupings to ration formulations, where the ration formulation is defined by a mapping from the
+        RuFaS ID of a feed to the percentage it makes up in a ration.
+
+    """
+
+    _om = OutputManager()
+    user_defined_rations: dict[AnimalCombination, dict[RUFAS_ID, float]]
+
+    @classmethod
+    def set_user_defined_rations(cls, ration_config: list[dict[str, Any]]) -> None:
+        """
+        Maps the input user-defined rations to Animal combinations.
+
+        Parameters
+        ----------
+        ration_config : list[dict[str, Any]]
+            List of dictionaries containing the user-defined rations for each animal combinantion.
+
+        """
+        info_map = {"class": cls.__class__.__name__, "function": cls.set_user_defined_rations.__name__}
+
+        cls.user_defined_rations = {
+            AnimalCombination.CALF: {},
+            AnimalCombination.GROWING: {},
+            AnimalCombination.CLOSE_UP: {},
+            AnimalCombination.LAC_COW: {},
+        }
```

Comment:
```suggestion
        cls.user_defined_rations = {ac: {} for ac in AnimalCombination}
```
Not sure if we specifically don't want user_defined_rations for the other AnimalCombinations but this is cleaner if we do.